tests/manage/z_cluster/nodes/test_node_replacement_proactive.py

```diff
@@ -232,3 +235,41 @@ def test_nodereplacement_proactive(self):
         assert ceph_cluster_obj.wait_for_rebalance(
             timeout=1800
         ), "Data re-balance failed to complete"
+
+
+@tier4b
+@skipif_no_lso
+@ignore_leftovers
+@bugzilla("1840539")
+class TestNodeReplacementLSO(ManageTest):
+    """
+    Node replacement twice on LSO:
+    node_x -> node_y
+    node_z -> node_x
+
+    After node_replacement, the deleted node (node_x) suppose to be removed from the ceph-osd-tree.
+    The BZ deals with the SECOND node_replacement.
+    The existence of the deleted node (node_x from previous replacement) in the crash-map ends with:
+      1. node is labeled for rack correctly
+      2. ceph side host still on the old rack
+    """
+
+    def test_nodereplacenet_lso(self):
+        node_name_to_delete = None
+        for i in range(2):
+            # Get random node name for replacement
+            node_name_to_delete = select_osd_node_name()
+            log.info(f"Selected node for replacement: {node_name_to_delete}")
+            delete_and_create_osd_node(node_name_to_delete)
+            ct_pod = pod.get_ceph_tools_pod()
+            tree_output = ct_pod.exec_ceph_cmd(ceph_cmd="ceph osd tree")
+            hosts_ids = tree_output["nodes"][0]["children"]
+            hosts_objs = [
+                obj
+                for obj in tree_output["nodes"]
+                if obj["type"] == "host" and obj["id"] in hosts_ids
+            ]
+            for host in hosts_objs:
+                assert (
+                    host["name"] not in node_name_to_delete
+                ), f"Deleted host {node_name_to_delete} still exist in ceph osd tree after node replacement"
```

Comment:
How are we making sure that  in the second node replacement, we are picking the node that was replaced first?
Here https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/node.py#L914 this function will return the newly created node name removing the old node, so in the second iteration looks like it it may pick the existing nodes in the cluster and not the node that was replaced first.

Reply:
The BZ deals with the situation where the deleted node still exists in the crash map.
IIUC, the approach to make sure that the deleted node is no longer in the crash map is "stronger" because it covers both cases where the newly created node is actually _new_ or the _old_ one.